datafusion/sqllogictest/test_files/union_by_name.slt

```diff
@@ -244,13 +244,14 @@ SELECT x, y FROM t1 UNION BY NAME (SELECT y, z FROM t2 INTERSECT SELECT 2, 2 as
 3 3 NULL
 NULL 2 2
 
-query III
+# TODO: Test regression on a new feature, with the same SanityCheckPlan failures as currently existing for this feature.
+query error
 SELECT x, y FROM t1 UNION ALL BY NAME (SELECT y, z FROM t2 INTERSECT SELECT 2, 2 as two FROM t1 ORDER BY 1) ORDER BY 1;
 ----
-1 1 NULL
-3 3 NULL
-3 3 NULL
-NULL 2 2
+DataFusion error: SanityCheckPlan
+caused by
```

Comment:
In this file, there are 3 other SanityCheckPlan failures in the same (newly added) feature.

I've been able to isolate this to the EnforceSorting, `pushdown_sort` subrule. I'm still in the process of figuring out the fix, hopefully will have something to share next week.

But IMO this is not a blocker. Feedback?